iceoryx_posh/doc/chunk_header.md

```diff
@@ -0,0 +1,218 @@
+# Summary
+
+Chunks are the transport capsules in iceoryx. They store data from a publisher as payload and are sent to one or more subscriber. Furthermore, there is some meta information which is stored alongside the payload, e.g. the size and the origin of the chunk. This data is composed in the `ChunkHeader` and located at the front of the chunk. Custom meta information can be added to extend the data from the `ChunkHeader` and tailor the chunk to specific use cases. While this makes the chunk layout more complex, this complexity would otherwise be distributed at different locations all over the code base, e.g. in the request/response feature. Additionally, the adjustments for the custom header makes arbitrary alignments of the payload trivial to implement.
+
+# Terminology
+
+| Name              | Description                                              |
+| :---------------- | :------------------------------------------------------- |
+| Chunk             | a piece of memory                                        |
+| Chunk Header      | contains meta information related to the chunk           |
+| Custom Header     | contains custom meta information, e.g. timestamps        |
+| Payload           | the user data                                            |
+| Back-Offset       | offset stored in front of the payload to calculate back to the chunk header |
+
+# Design
+
+## Considerations
+
+- it's not uncommon to record chunks for a later replay -> detect incompatibilities on replay
+- iceoryx runs on multiple platforms -> endianness of recorded chunks might differ
```

Comment:
btw. we could store an endianness indicator in the `ChunkHeader`, e.g. by replacing `m_reserved2` and `m_reserved3` with a `uint16_t m_endianness = 0xA55A`. On big endian platforms the memory would be `... m_chunkHeaderVersion m_reserved1 0xA5 0x5A` and on little endian `... m_chunkHeaderVersion m_reserved1 0x5A 0xA5`. Alternatively, one byte only could be used as flag.
I just don't know if it makes sense to store it in the `ChunkHeader` or shift the responsibility to the record framework

Reply:
Good point @sculpordwarf  @frank-schmidtner What do you think, should this be take care of by the recording framework? Any preferences?